service/resources/google_sheets.py

```diff
@@ -61,6 +61,9 @@ def on_get(self, _req, resp):
 
             column_idx = gspread.utils.a1_to_rowcol(column + '1')[1]
             cells_found = worksheet.findall(value, in_column=column_idx)
+            if len(cells_found) == 0:
+                raise gspread.exceptions.CellNotFound
```

Comment:
If the worksheet has one bad submission, will the on_get return an empty set or a set without the bad submission?

Reply:
what do you mean by "bad submission"?  This area of code is querying for a value in a given column.  If no values are found (ie. lengh of array ==0) then we return a 404 error.